lodash/lodash-tests.ts

```diff
@@ -2245,6 +2245,46 @@ namespace TestUniqBy {
     }
 }
 
+// _.uniqWith
+namespace TestUniqWith {
+    type SampleObject = {a: number; b: string; c: boolean};
+
+    let array: any[];
+    let objArray: SampleObject[];
+    let letters: string;
+    let comparator: (a: any, b : any) => boolean;
+
+    {
+        let result: any[];
+        result = _.uniqWith(array, comparator);
+    }
+
+    {
+        let result: SampleObject[];
+        result = _.uniqWith(objArray, comparator);
+    }
+
+    {
+        let result: string[];
+        result = _.uniqWith('abc', comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _(['abc', 'bca']).uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').chain().uniqWith<string[]>(comparator);
```

Comment:
I believe you're referring to type assertions, but these are [type arguments](https://www.typescriptlang.org/docs/handbook/generics.html).